mediacapture-streams/MediaStreamTrack-applyConstraints.https.html

```diff
@@ -8,51 +8,46 @@
 
   // https://w3c.github.io/mediacapture-main/#dom-mediastreamtrack-applyconstraints
 
-  promise_test(t => {
-    return navigator.mediaDevices.getUserMedia({ video: true })
-      .then(t.step_func(stream => {
-        return stream.getVideoTracks()[0].applyConstraints(
-          { groupId: { exact: "INVALID" } }).then(
-            t.unreached_func('Accepted invalid groupID'),
-            t.step_func(e => {
-              assert_equals(e.name, 'OverconstrainedError');
-              assert_equals(e.constraint, 'groupId');
-            }));
-      }));
+  promise_test(async t => {
+    const stream = await navigator.mediaDevices.getUserMedia({ video: true });
+    const [track] = stream.getVideoTracks();
+    t.add_cleanup(() => track.stop());
+    try {
+      await track.applyConstraints({ groupId: { exact: "INVALID" } });
+      t.unreached_func('Accepted invalid groupID')();
+    } catch (e) {
+      assert_equals(e.name, 'OverconstrainedError');
+      assert_equals(e.constraint, 'groupId');
+    }
   }, 'applyConstraints rejects invalid groupID');
 
-  promise_test(t => {
-    return navigator.mediaDevices.getUserMedia({ video: true })
-      .then(t.step_func(stream => {
-        var track = stream.getVideoTracks()[0];
-        var groupId = track.getSettings().groupId;
-        return track.applyConstraints({ groupId: "INVALID" }).then(
-          t.step_func(() => {
-            assert_equals(track.getSettings().groupId, groupId);
-          }));
-      }));
+  promise_test(async t => {
+    const stream = await navigator.mediaDevices.getUserMedia({ video: true });
+    const [track] = stream.getVideoTracks();
+    t.add_cleanup(() => track.stop());
+    const groupId = track.getSettings().groupId;
```

Comment:
Maybe assert that `groupId` is not `undefined`?

Reply:
I will fix it this PR.